File: executables/checkDataset.cpp
#include <cstdlib>
#include <iostream>
#include "chm_tools/Dataset.hpp"
namespace fs = chm::fs;

#ifndef REPO_DIR
	constexpr auto REPO_DIR = "";
#endif

int main() {
	try {
		const auto dataDir = fs::path(REPO_DIR) / "data";
		const auto datasetPath = dataDir / "test.bin";
		const auto descPath = dataDir / "testCpp.txt";
		chm::Dataset(datasetPath).writeLongDescription(descPath);
		std::cout << "Description of dataset " << datasetPath << " written to " << descPath << ".\n";

	} catch(const std::exception& e) {
		std::cerr << "[ERROR] " << e.what() << '\n';
		return EXIT_FAILURE;
	}
	return EXIT_SUCCESS;
}

File: src/scripts/checkDataset.py
from chmDataset import runner
from chmDataset.AppError import AppError
from chmDataset.Dataset import Dataset
from pathlib import Path

N = "\n"

def main():
	try:
		dataDir = Path(__file__).parents[1] / "data"
		datasetPath = dataDir / "test.hdf5"
		descPath = dataDir / "testPy.txt"
		Dataset.fromHDF(datasetPath).writeDescription(descPath)
		print(f"Description of dataset {datasetPath} written to {descPath}.")

	except FileNotFoundError:
		raise AppError("HDF5 dataset not found. Run datasetGenerator.py first.")

if __name__ == "__main__":
	runner.run(main)
